### PR TITLE
EVG-6159: restart system failed tasks

### PR DESCRIPTION
**Pull request by ybrill**
Restarting tasks checks that the status is a finished status (either "failed" or "succeeded"). However, for display tasks if an execution task system-failed the status would be "system-failed", so it would not be restarted.
This PR is to make exec task and display task statuses consistent.

### CODE REVIEW

File: model/task/task.go [hunk 1]
```diff
@@ -769,28 +769,28 @@ func (t *Task) MarkEnd(finishTime time.Time, detail *apimodels.TaskEndDetail) er
 
 }
 
-func displayTaskPriority(status string) int {
-	switch status {
-	case evergreen.TaskStarted:
-		return 10
-	case evergreen.TaskUndispatched:
-		return 40
+func displayTaskPriority(t Task) int {
```
Comment (tychoish) [inline model/task/task.go]: make this a method on task.

Comment (ybrill) [reply model/task/task.go]: done.

Comment (john-m-liu) [reply model/task/task.go]: I disagree. This makes us keep around potentially large slices of task documents when we only need the status

Comment (ybrill) [reply model/task/task.go]: But if we're sorting tasks there must be a slice of tasks somewhere.

**Comment by ybrill** [comment]
That's a good question.
- I don't think we need to for the purposes of the bug this PR is addressing (it's unlikely someone will want to restart an old display task).
-  I don't think any of the changes in the PR will break existing display tasks.

**Pull request merged**